eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -773,28 +773,30 @@ private class InteractiveTxBuilder(replyTo: ActorRef[InteractiveTxBuilder.Respon
     if (unsignedTx.localInputs.isEmpty) {
       context.self ! SignTransactionResult(PartiallySignedSharedTransaction(unsignedTx, TxSignatures(fundingParams.channelId, tx, Nil, sharedSig_opt)))
     } else {
-      // we only sign our wallet inputs, and check that we can spend our wallet outputs
-      val ourWalletInputs = unsignedTx.localInputs.map(i => tx.txIn.indexWhere(_.outPoint == OutPoint(i.previousTx, i.previousTxOutput.toInt)))
-      val ourWalletOutputs = unsignedTx.localOutputs.collect {
-        // README!! there are 2 types of local outputs:
-        // Change, which go back into our wallet
-        // NonChange, which go to an external address (typically during a splice-out)
-        // Here we only keep outputs which are ours i.e go back into our wallet
-        // And we trust that NonChange outputs are valid. This only works if the entry point for creating such outputs is trusted (for example, a secure API call)
+      // We only sign our wallet inputs, and check that we can spend our wallet outputs.
+      val ourWalletInputs = unsignedTx.localInputs.map(i => tx.txIn.indexWhere(_.outPoint == i.outPoint)).filter(_ >= 0)
+      val ourWalletOutputs = unsignedTx.localOutputs.map {
         case Output.Local.Change(_, amount, pubkeyScript) => tx.txOut.indexWhere(output => output.amount == amount && output.publicKeyScript == pubkeyScript)
-      }
+        // Non-change outputs may go to an external address (typically during a splice-out).
+        // Here we only keep outputs which are ours i.e explicitly go back into our wallet.
+        // We trust that non-change outputs are valid: this only works if the entry point for creating such outputs is trusted (for example, a secure API call).
+        case _: Output.Local.NonChange => -1
+      }.filter(_ >= 0)
```

Comment:
Same here: all local change outputs should be found in the tx ?

Reply:
Removed in https://github.com/ACINQ/eclair/pull/2726/commits/e25ce20eac87a0ccadd94d1975094382cc12f9ab